### PR TITLE
add psc-ide for purescript

### PR DESCRIPTION
**Pull request by jeizsm**

### CODE REVIEW

File: layers/+lang/purescript/packages.el [hunk 1]
```diff
@@ -40,3 +41,23 @@
         "msi" 'psci
         "msm" 'psci/load-module!
         "msp" 'psci/load-project-modules!))))
+
+(when (configuration-layer/layer-usedp 'auto-completion)
+  (defun purescript/post-init-company ()
```
Comment (TheBB) [inline layers/+lang/purescript/packages.el]: If you have this, you also need to list `company` as a package.

File: layers/+lang/purescript/packages.el [hunk 2]
```diff
@@ -40,3 +41,23 @@
...
+    (spacemacs|add-company-hook purescript-mode))
+
+  (defun purescript/init-psc-ide ()
+    (use-package psc-ide
+      :if (configuration-layer/package-usedp 'company)
+      :defer t
+      :init
+      (push 'psc-ide company-backends-purescript-mode)
+      (progn
+        (add-hook 'purescript-mode-hook 'psc-ide-mode))
```
Comment (TheBB) [inline layers/+lang/purescript/packages.el]: Please put both these in a single `progn`.

File: layers/+lang/purescript/config.el [hunk 3]
```diff
@@ -12,3 +12,6 @@
 
 
 ;; Nothing here yet
+(spacemacs|defvar-company-backends purescript-mode)
+(defvar psc-ide-executable "psc-ide")
+(defvar psc-ide-server-executable "psc-ide-server")
```
Comment (TheBB) [inline layers/+lang/purescript/config.el]: Any reason you have defined these variables?

Comment (jeizsm) [inline layers/+lang/purescript/config.el]: Yes. It is needed for set path to psc-ide executables

Comment (TheBB) [inline layers/+lang/purescript/config.el]: Wouldn't it be better to set their value in `init-psc-ide` instead, then?

Comment (TheBB) [inline layers/+lang/purescript/config.el]: Besides you have just given them the default values they already have in the package...

Comment (jeizsm) [inline layers/+lang/purescript/config.el]: I thought that it is necessary that the user can override them in dotspacemacs-configuration-layers

Comment (TheBB) [inline layers/+lang/purescript/config.el]: No, no, that will work fine without this.

**Comment by jeizsm** [comment]
Please review again

File: layers/+lang/purescript/config.el [hunk 4]
```diff
@@ -12,3 +12,4 @@
 
 
 ;; Nothing here yet
```
Comment (StreakyCobra) [inline layers/+lang/purescript/config.el]: This comment can probably be removed now :-)

Comment (jeizsm) [inline layers/+lang/purescript/config.el]: Definitely :)
Done

**Pull request closed without merging**